frontend/src/components/job_form/job_form.jsx

```diff
@@ -0,0 +1,138 @@
+import { BasicDateTimePicker } from "./time_picker"
+import { Container, Form, Button } from 'react-bootstrap';
+import { useState } from 'react';
+import toast from 'react-hot-toast';
+
+export const JobForm = () => {
+  const [jobTitle, setJobTitle] = useState('');
+  const [jobType, setJobType] = useState('');
+  const [price, setPrice] = useState(0);
+  const [timeEstimate, setTimeEstimate] = useState(0);
+  const [address, setAddress] = useState('');
+  const [error, setError] = useState('');
+  const [startTime, setStartTime] = useState(new Date());
+  const [endTime, setEndTime] = useState(new Date());
+
+  const handleTimeChange = (e, timeFrame) => {
+    if (timeFrame === "start") {
+      setStartTime(e);
+    }
+
+    if (timeFrame === "end") {
+      setEndTime(e);
+    }
+  } 
+
+  const validateForm = () => {
+    let currUnix = Math.floor(new Date().getTime() / 1000);
+    let startUnix = Math.floor(startTime.getTime() / 1000);
+    let endUnix = Math.floor(endTime.getTime() / 1000);
+
+    if (!jobTitle) {
+      setError('Job title is required');
+      return false;
+    }
+
+    if (!jobType) {
+      setError('Job type is required');
+      return false;
+    }
+
+    if (!price) {
+      setError('Compensation is required');
+      return false;
+    }
+
+    if (!timeEstimate) {
+      setError('Time estimate is required');
+      return false;
+    }
+
+    if (!address) {
+      setError('Address is required');
+      return false;
+    }
+
+    if (startUnix < currUnix) {
+      setStartTime(new Date());
+    }
+    
+    if (endUnix < startUnix || endUnix < currUnix) {
+      setError('Invalid listing dates and/or times');
+      return false;
+    }
+
+    if (endUnix - startUnix < 172800) {
+      setError('Listing time too short, must be up for at least 2 days')
```

Comment:
I think a minimum length of 2 days might be a little too long. Maybe something like 3*timeEstimate?
I also think there should be a maximum availability of around 4 days; having a length longer than a week may cause bugs in the availability filtering.